// python3 -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. agenda.proto
syntax = "proto3";

service Manager {
    rpc AddPerson(Person) returns (BooleanReply) {}
    rpc DelPerson(PersonId) returns (BooleanReply) {}
    rpc SearchPersonName(PersonName) returns (stream Person) {}
    rpc SearchPersonId(PersonId) returns (PersonReply) {}
    rpc ListContacts(ContactsRequest) returns (stream Person) {}
}

message Person {
    string name = 1;
    int32 id = 2;
    string email = 3;

    enum PhoneType {
        MOBILE = 0;
        HOME = 1;
        WORK = 2;
    }

    message PhoneNumber {
        string number = 1;
        PhoneType type = 2;
    }

    repeated PhoneNumber phones = 4;
}

message PersonId {
    int32 id = 1;
}

message PersonName {
    string name = 1;
}

message PersonReply {
    Person person = 1;
}

message BooleanReply {
    bool reply = 1;
}

message ContactsRequest {
    bool listContacts = 1;
}